python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -55,79 +65,334 @@ def __init__(
+            for linestring in geom:
+                linestrings.append(
+                    LineString([tuple(child) for child in linestring])
+                )
+            return MultiLineString(linestrings)
+
+    def _polygon_to_shapely(self, geom):
+        if len(geom) == 1:
+            rings = []
+            for ring in geom[0]:
+                rings.append(tuple(tuple(point) for point in ring))
+            return Polygon(rings[0], rings[1:])
+        else:
+            polygons = []
+            for p in geom:
+                rings = []
+                for ring in p:
+                    rings.append(tuple([tuple(point) for point in ring]))
+                polygons.append(Polygon(rings[0], rings[1:]))
+            return MultiPolygon(polygons)
+
+    @cached_property
+    def _arrow_to_shapely(self):
+        type_map = {
+            Feature_Enum.POINT: Point,
+            Feature_Enum.MULTIPOINT: MultiPoint,
+            Feature_Enum.LINESTRING: self._linestring_to_shapely,
+            Feature_Enum.POLYGON: self._polygon_to_shapely,
+        }
+        return type_map
+
+    def to_shapely(self):
+        # Copy the GPU data to host for iteration and deserialization
+        result_types = self._column._meta.input_types.to_arrow()
+        union_types = self._column._meta.input_types
+
+        # Get the shapely serialization methods we'll use here.
+        shapely_fns = [
+            self._arrow_to_shapely[Feature_Enum(x)]
+            for x in result_types.to_numpy()
+        ]
+
+        union = self.to_arrow()
+
+        index = range(0, len(self))
+        # Convert the iterable types into a predictable format for the
+        # upcoming zip
+        if isinstance(index, slice):
+            start, stop, step = index.indices(len(union_types))
+            indexes = range(0, max(stop - start, 1), step)
+            # Use the reordering _column._data if it is set
+            indexes = (
+                self._column._data[indexes].to_pandas()
+                if self._column._data is not None
+                else indexes
+            )
+            serialization_functions = pd.Series(shapely_fns)[indexes]
+        elif isinstance(index, Iterable):
+            indexes = list(range(len(list(index))))
+            # Use the reordering _column._data if it is set
+            indexes = (
+                self._column._data[indexes].to_pandas()
+                if self._column._data is not None
+                else indexes
+            )
+            serialization_functions = pd.Series(shapely_fns)[indexes]
+        else:
+            indexes = [index]
+            serialization_functions = [shapely_fns[indexes]]
+
+        # Serialize to Shapely!
+        results = []
+        for (result_index, shapely_serialization_fn) in zip(
+            indexes, serialization_functions
+        ):
+            results.append(
+                shapely_serialization_fn(union[result_index].as_py())
+            )
```

Comment:
Since `to_shapely` discards all index information, the `if-elses` here are basically unnecessary. 

```suggestion
        # Serialize to Shapely!
        results = []
        for (result_index, shapely_serialization_fn) in zip(
             range(0, len(self)), serialization_functions
        ):
            results.append(
                shapely_serialization_fn(union[result_index].as_py())
            )
```

Reply:
nice!